lix/packages/sdk/src/open/openLix.ts

```diff
@@ -149,22 +149,90 @@ async function loadPlugins(db: Kysely<LixDatabase>) {
 // 	}
 // }
 
+async function getChangeHistory({
+	atomId,
+	depth,
+	fileId,
+	pluginKey,
+	diffType,
+	db,
+}: {
+	atomId: string
+	depth: number
+	fileId: string
+	pluginKey: string
+	diffType: string
+	db: Kysely<LixDatabase>
+}): Promise<any[]> {
+	if (depth > 1) {
+		throw new Error("depth > 1 not supported yet")
+	}
+
+	const { commit_id } = await db
+		.selectFrom("ref")
+		.select("commit_id")
+		.where("name", "=", "current")
+		.executeTakeFirstOrThrow()
+
+	let nextCommitId = commit_id
+	let firstChange
+	while (!firstChange && nextCommitId) {
+		const commit = await db
+			.selectFrom("commit")
+			.selectAll()
+			.where("id", "=", nextCommitId)
+			.executeTakeFirstOrThrow()
+
+		nextCommitId = commit.parent_id
+
+		firstChange = await db
+			.selectFrom("change")
+			.selectAll()
+			.where("commit_id", "=", commit.id)
+			.where((eb) => eb.ref("value", "->>").key("id"), "=", atomId)
+			.where("plugin_key", "=", pluginKey)
+			.where("file_id", "=", fileId)
+			.where("type", "=", diffType)
+			.executeTakeFirst()
+
+		console.log(commit, firstChange)
```

Comment:
i assume the console.log was not intended

Reply:
@samuelstroschein  ahh i was looking for that one !